### ISSUE TITLE
Improvement of accessibility (contrasting colors)

### ISSUE THREAD

**Issue by joshuai96** [jitsi/jitsi-meet#15383]
### What problem are you trying to solve?

We are currently trying to improve Jitsi's accessibility after a customer had their Jitsi instance checked for conformance to the WCAG 2.

This issue is a excerpt of the test and serves as a discussion about ways to handle the following points.

For [Success Criterion 1.4.3 Contrast (Minimum)](https://www.w3.org/TR/WCAG22/#contrast-minimum) there are some components which don't fulfill the required contrast of 4,5:1.

- The white text on the grey background, when you didn't give access to the microphone and camera has a contrast of 4,1:1.
![image](https://github.com/user-attachments/assets/898981c7-df7f-4944-89a3-929055e6757b)
- The red "Reject" text has a contrast of 3,5:1.
![image](https://github.com/user-attachments/assets/5e3570fd-0743-4844-9b37-eaa1d9854ae7)
- Some of the user avatars (colored circle with initials) have bad contrasting colors.
- Most placeholder texts have a contrast of 2,9:1
![image](https://github.com/user-attachments/assets/ad11aff9-f22b-45c5-b3fc-98111c20e668)
![image](https://github.com/user-attachments/assets/4650ebe1-a5b7-4828-9300-d8bc6216d402)
- Background "Half blur" has a contrast of 2,5:1.
![image](https://github.com/user-attachments/assets/fb47b67c-7b47-4116-9bc5-a17279dd8a1e)

For [Success Criterion 2.4.7 Focus Visible](https://www.w3.org/TR/WCAG22/#focus-visible) we already supplied two PRs, but there is still an open point:

- For the selected and unselected category the contrast is 1,4:1
![image](https://github.com/user-attachments/assets/15a1b314-054e-465f-b188-f6d1119cf832)

For [Success Criterion 1.4.11 Non-text Contrast](https://www.w3.org/TR/WCAG22/#non-text-contrast) there are some components which don't fulfill the required contrast of 3:1.

- The white "perfomance" icon on green background has a contrast of 2,3:1.
![image](https://github.com/user-attachments/assets/845e347d-5d8e-469f-a9ab-c38378565e1f)

- Active and inactive toolbar buttons have a contrast of 1,7:1.
![image](https://github.com/user-attachments/assets/49e24570-95a3-47e2-9fc9-505aabd2befd)

- All input fields (see above for chat and poll) have a contrast of 1,7:1 to the background.
![image](https://github.com/user-attachments/assets/e228d712-0a0a-4aed-9436-7ba431162582)

### What solution would you like to see?

I am by far no designer and I don't know if the colors used in Jitsi have some kind of corporate identity (CI) or corporate design (CD) behind them.

I would like if you could supply alternate colors that fix the contrast issues.

### Is there an alternative?

If there is no CI or CD behind the used colors, we could suggest new colors that fix the aforementioned contrast issues.

**Comment by saghul** [comment]
Thanks for the report! We have work planned around WCAG compliance for Q1 2025, I'll make sure these make it to the list. Cheers!

**Comment by joshuai96** [comment]
We have some more test results regarding the WCGA, we're trying to improve. Is there any way to see or add to your list of planed changes.  It might be helpful to so we may contribute as we plan to improve our test results in Q1 2025 also, so no double work on the same stuff might happen.

**Comment by saghul** [comment]
Work will start in the new year and we'll get collaborations from another internal team, so coordination is a bit weird here.

That said, since both contributions will need to be made here, to this repo, I think you can just get started with the low langing fruit, or things that don't require design decissions, and we can take it from there.

**Comment by joshuai96** [comment]
Hello and a late new year.

Have there been any updates concerning accessibility and WCAG  compliance?

We are starting to work on the mentioned WCAG compliance tests and trying to reduce duplication, if topics may have already been handled.

**Comment by saghul** [comment]
@raduanastase8x8 a fellow 8x8 teammate is working on it. Feel free to drop questions here.

**Comment by joshuai96** [comment]
> I think this is in a grey area because the selection indicator (the blue line) is meeting the contrast threshold and the user can differentiate between the two. I agree that the best solution would be to have the background meet the threshold as well, but I'm not sure that I will get the design team on board.
> 
> > * For the selected and unselected category the contrast is 1,4:1
> >   ![image](https://github.com/user-attachments/assets/15a1b314-054e-465f-b188-f6d1119cf832)

That is true, like the tool bar, the difference between active and inactive was flagged by the compliance test.


> This is a single image, the white shapes don't offer important information, so I am pretty sure this will pass.
> 
> > * The white "perfomance" icon on green background has a contrast of 2,3:1.
> >   ![image](https://github.com/user-attachments/assets/845e347d-5d8e-469f-a9ab-c38378565e1f)

Even though they convey no important information, the contrasting was flagged by the institution conducting the compliance test. I, for myself, understand that the (changing) color of that button is more important to convey the meaning. I'll try to reason on this with them, but I think it's just a good idea to have it mentioned here.

> Not sure if we'll change the background, but we need an indicator, like the blue line above.
> 
> > * Active and inactive toolbar buttons have a contrast of 1,7:1.
> >   ![image](https://github.com/user-attachments/assets/49e24570-95a3-47e2-9fc9-505aabd2befd)

As with the settings screen above, this may not be sufficient enough, as it got flagged also. IIRC somewhere, the size of such indicators might play a role also (area of indicator to area of button).

> The input background doesn't need to meet the 3:1 contrast, but we should have something that meets the contrast, like a border. Like in this example https://www.w3.org/WAI/tutorials/forms/validation/
> 
> > * All input fields (see above for chat and poll) have a contrast of 1,7:1 to the background.
> >   ![image](https://github.com/user-attachments/assets/e228d712-0a0a-4aed-9436-7ba431162582)

Isn't the blue border implemented on every input element?

Thank you for taking your time, keeping me informed :)

**Comment by joshuai96** [comment]
TWIMC
to meet our requirements for a second WCAG compliance test, we changed some colors slightly and added some borders on places where changing colors would lead to bad contrast ratios with surrounding colors.

As CI/CD colors is a topic that may need some more decision, we could at least propose our changes in a PR, if wished for.

**Comment by raduanastase8x8** [comment]
I think a PR would be usefull. Thanks!
